plugins/obs-webrtc/webrtc/src/output.rs

```diff
@@ -0,0 +1,383 @@
+
+                                    let stats = &mut stats.write().unwrap();
+                                    if let Some(StatsReportType::Transport(transport_stats)) = pc_stats.reports.get("ice_transport") {
+                                        stats.bytes_sent = transport_stats.bytes_sent as u64;
+                                    }
+                                    Ok(())
+                                }
+                            };
+
+                           if let Err(e) = result {
+                                break 'worker_loop e;
+                           }
+                        }
+                    }
+                );
+
+                match result {
+                    WorkerResult::Close => {}
+                    WorkerResult::Error(e) => {
+                        if let Some(callback) = &mut *error_callback.lock().unwrap() {
+                            error!("Worker encountered error: {e:?}");
+                            callback(e);
+                        }
+                    }
+                }
+
+                info!("Exiting worker thread");
+            }
+        });
+
+        *self.worker_handle.lock().unwrap() = Some(worker_handle);
+    }
+
+    pub async fn new() -> Result<Self> {
+        let video_track = Arc::new(TrackLocalStaticSample::new(
+            RTCRtpCodecCapability {
+                mime_type: MIME_TYPE_H264.to_owned(),
+                clock_rate: 90000,
+                sdp_fmtp_line: "profile-level-id=428014; max-fs=3600; max-mbps=108000; max-br=1400"
+                    .to_string(),
+                ..Default::default()
+            },
+            "video".to_owned(),
+            "webrtc-rs".to_owned(),
+        ));
+
+        let audio_track = Arc::new(TrackLocalStaticSample::new(
+            RTCRtpCodecCapability {
+                mime_type: MIME_TYPE_OPUS.to_owned(),
+                ..Default::default()
+            },
+            "audio".to_owned(),
+            "webrtc-rs".to_owned(),
+        ));
+
+        let mut m = MediaEngine::default();
+        m.register_default_codecs()?;
+
+        let mut registry = Registry::new();
+        registry = register_default_interceptors(registry, &mut m)?;
+
+        let gather_ips = if_addrs::get_if_addrs()
+            .unwrap_or_else(|_| Vec::new())
+            .iter()
+            .fold(Vec::new(), |mut gather_ips, addr| {
+                if !addr.is_loopback() {
+                    trace!(
+                        "Valid gather address for interface {}: {:#?}",
+                        addr.name,
+                        addr.ip()
+                    );
+                    gather_ips.push(addr.ip());
+                } else {
+                    trace!(
+                        "Loopback address ignored for interface {}: {:#?}",
+                        addr.name,
+                        addr.ip()
+                    );
+                }
+                gather_ips
+            });
```

Comment:
```suggestion
        let gather_ips: Vec<_> = if_addrs::get_if_addrs()
            .unwrap_or_else(|_| Vec::new())
            .into_iter()
            .filter(|addr| {
                if !addr.is_loopback() {
                    trace!(
                        "Valid gather address for interface {}: {:#?}",
                        addr.name,
                        addr.ip()
                    );
                    true
                } else {
                    trace!(
                        "Loopback address ignored for interface {}: {:#?}",
                        addr.name,
                        addr.ip()
                    );
                    false
                }
            })
            .map(|addr| addr.ip())
            .collect();
```

Reply:
Fixed in 062d2fa857d376f7940fbc1ea0f9da0f3e8d436d